expenses-manager-backend/src/main/java/com/endava/expensesmanager/controller/CategoryController.java

```diff
@@ -0,0 +1,32 @@
+package com.endava.expensesmanager.controller;
+
+import com.endava.expensesmanager.entity.Category;
+import com.endava.expensesmanager.service.CategoryService;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/categories")
+public class CategoryController {
+    private final CategoryService categoryService;
+
+    public CategoryController(CategoryService categoryService) {this.categoryService = categoryService;}
+
+    @GetMapping
+    public ResponseEntity<List<String>> getAllCategoryDescriptions() {
```

Comment:
Here it's not really ok to get a list of descriptions and then get a category by description. What if you have 2 categories with the same description?

I suggest that you have an endpoint that returns a list of categories in which you have the id of the category, that you know is unique, then you can make an endpoint get category by id.